asciidoctorj-core/src/main/java/org/asciidoctor/jruby/internal/RubyHashMapDecorator.java

```diff
@@ -22,9 +22,35 @@
 
     private RubyClass abstractNodeClass;
 
+    private Class rubyKeyType;
+
     public RubyHashMapDecorator(RubyHash rubyHash) {
+        this(rubyHash, RubySymbol.class);
+    }
+
+    /**
+     * Wrap a RubyHash map that uses a key of type `rubyKeyType`.
+     *
+     * Regardless of `rubyKeyType`, the wrapper will always use Java String
+     * as the exposed key type and coerce internally as appropriate.
+     *
+     * @param rubyHash hash map to wrap
+     * @param rubyKeyType key type, valid values: RubySymbol.class or String.class
+     */
+    public RubyHashMapDecorator(RubyHash rubyHash, Class rubyKeyType) {
         this.rubyRuntime = rubyHash.getRuntime();
         this.rubyHash = rubyHash;
+        if (rubyKeyType != RubySymbol.class && rubyKeyType != String.class) {
+            throw new UnsupportedOperationException("key type must be either RubySymbol or String");
+        }
+        this.rubyKeyType = rubyKeyType;
+    }
+
+    private Object coerceKey(Object key) {
```

Comment:
Is there anything speaking against making `key` a `String`?

Reply:
Right you are! I think the method signature could and should be `private Object coerceKey(String key)`

Reply:
Oh... I remember now, I left it as `Object` to avoid casting from all the callers.  

Lemme know what you prefer.  I have no strong opinion.

Reply:
I found it easier to reason about this method when I saw that the key is a String. (Would have been nice if Java had a variant type, then the signature would have been even clearer, but that doesn't exist afaik)
But I guess this is fine for now a the method is private.

Reply:
Hey, it is easy for me to switch it to String. I'll go ahead and do so.